app/services/invite_provider_user.rb

```diff
@@ -0,0 +1,60 @@
+require 'jwt'
+
+class InviteProviderUser
+  def initialize(provider_user_form:)
+    @provider_user_form = provider_user_form
+  end
+
+  def call
+    ActiveRecord::Base.transaction do
+      if @provider_user_form.save
+        response = invite_user_to_dfe_sign_in
+        raise DfeSignInApiError.new(response) if response && !response.status.success?
```

Comment:
Feels like we're mixing concerns in this method. Would the `raise` be better in the `invite_user_to_dfe_sign_in` method?

Reply:
Ok. I guess I was enjoying calling `invite_user_to_dfe_sign_in` directly for troubleshooting error cases too much. Will keep this as a public method, as it's good for troubleshooting or implementing a 're-send the invite' button.